src/cpp/multisolver_interface/SolverClp.cpp

```diff
@@ -35,23 +35,14 @@ SolverClp::SolverClp()
     set_output_log_level(0);
 }
 
-SolverClp::SolverClp(const SolverAbstract& toCopy):
+SolverClp::SolverClp(const SolverClp& toCopy):
     SolverClp()
 {
-    // Try to cast the solver in fictif to a SolverClp
-    if (const auto c = dynamic_cast<const SolverClp*>(&toCopy))
-    {
-        _clp = ClpSimplex(c->_clp);
-        _fp = c->_fp;
-        if (_fp)
-        {
-            _clp.messageHandler()->setFilePointer(c->_fp);
-        }
-    }
-    else
+    _clp = ClpSimplex(toCopy._clp);
+    _fp = toCopy._fp;
+    if (_fp)
     {
-        number_of_problems_counter() -= 1;
-        throw InvalidSolverForCopyException(toCopy.get_solver_name(), name_, LOGLOCATION);
+        _clp.messageHandler()->setFilePointer(toCopy._fp);
```

Comment:
No difference, but more clear in my opinion (set the "copied file pointer" to the "copied _clp")
```suggestion
        _clp.messageHandler()->setFilePointer(_fp);
```